src/test/java/BoardTest.java

```diff
@@ -66,4 +69,16 @@ public void BoardMirrorReversesRowsFixesColumns() {
         Assertions.assertEquals(board.getPieceAtMirrorCoords(0, 7 - 1), p1);
         Assertions.assertEquals(board.getPieceAtMirrorCoords(1, 7 - 3), p2);
     }
+
+    @Test
+    public void AdvanceCurrentPlayer() {
+        Board board = new Board();
+        Player p1 = new Player("Player 1", 'W');
+        Player p2 = new Player("Player 2", 'B');
+        Player[] players = {p1, p2};
+        board.setPlayers(players);
+        board.setCurrentPlayer(p1);
+        board.advanceCurrentPlayer();
+        Assertions.assertEquals(p2, board.getCurrentPlayer());
```

Comment:
Change to `Assertions.assertSame(p2, board.getCurrentPlayer())` as this also checks to see if they are refering to the same object.